x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/xmatters/xmatters_params.tsx

```diff
@@ -0,0 +1,141 @@
+      'xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.severitySelectCriticalOptionLabel',
+      {
+        defaultMessage: 'Critical',
+      }
+    ),
+  },
+  {
+    value: 'high',
+    text: i18n.translate(
+      'xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.severitySelectHighOptionLabel',
+      {
+        defaultMessage: 'High',
+      }
+    ),
+  },
+  {
+    value: 'medium',
+    text: i18n.translate(
+      'xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.severitySelectMediumOptionLabel',
+      {
+        defaultMessage: 'Medium',
+      }
+    ),
+  },
+  {
+    value: 'low',
+    text: i18n.translate(
+      'xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.severitySelectLowOptionLabel',
+      {
+        defaultMessage: 'Low',
+      }
+    ),
+  },
+  {
+    value: 'minimal',
+    text: i18n.translate(
+      'xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.severitySelectMinimalOptionLabel',
+      {
+        defaultMessage: 'Minimal',
+      }
+    ),
+  },
+];
+
+const XmattersParamsFields: React.FunctionComponent<ActionParamsProps<XmattersActionParams>> = ({
+  actionParams,
+  editAction,
+  index,
+  messageVariables,
+  errors,
+}) => {
+  useEffect(() => {
+    if (!actionParams) {
+      editAction(
+        'actionParams',
+        {
+          signalId: '{{rule.id}}:{{alert.id}}',
+          alertActionGroupName: '{{alert.actionGroupName}}',
+          ruleName: '{{rule.name}}',
+          date: '{{date}}',
+          spaceId: '{{rule.spaceId}}',
+        },
+        index
+      );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [actionParams]);
+  return (
+    <>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiFormRow
+            id="xmattersSeverity"
+            fullWidth
+            label={i18n.translate(
+              'xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.severity',
+              {
+                defaultMessage: 'Severity',
+              }
+            )}
+          >
+            <EuiSelect
+              fullWidth
+              data-test-subj="severitySelect"
+              options={severityOptions}
+              hasNoInitialSelection={isUndefined(actionParams.severity)}
+              value={actionParams.severity}
+              onChange={(e) => {
+                editAction('severity', e.target.value, index);
+              }}
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiFormRow
+            id="xmattersTags"
+            fullWidth
+            label={i18n.translate(
+              'xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.tags',
+              {
+                defaultMessage: 'Tags',
+              }
+            )}
+          >
+            <TextFieldWithMessageVariables
```

Comment:
Is it ok for tags to be a text field? Maybe the [Combo box](https://elastic.github.io/eui/#/forms/combo-box) is more appropiate.

Reply:
the expectation is for the user to use the {{rule.tags}} variable but add extra tags if they want, which is why I've used this text field type.

Reply:
Thank for the explanation! Wasn't aware of that.